keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+     * DELETE cannot be applied to an individual version of a secret. This operation requires the secrets/delete permission.
+     *
+     * @param name The name of the secret to be deleted.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the Deleted Secret instance of the secret that was deleted.
+     */
+    public Mono<RestResponse<DeletedSecret>> deleteSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.deleteSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Lists deleted secrets for the key vault.
+     * The Get Deleted Secrets operation returns the secrets that have been deleted for a vault enabled for soft-delete.
+     * This operation requires the secrets/list permission.
+     *
+     * @param name The name of the deleted secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the Deleted {@link Secret}.
+     */
+    public Mono<RestResponse<DeletedSecret>> getDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.getDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Permanently deletes the specified secret.
+     * The purge deleted secret operation removes the secret permanently, without the possibility of recovery.
+     * This operation can only be enabled on a soft-delete enabled vault. This operation requires the secrets/purge permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the RestResponse.
+     */
+    public Mono<RestResponse> purgeDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.purgeDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Recovers the deleted secret to the latest version.
+     * Recovers the deleted secret in the key vault. This operation can only be performed on a soft-delete enabled vault.
+     * This operation requires the secrets/recover permission.
+     *
+     * @param name The name of the deleted secret to be recovered.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the {@link Secret} that was recovered
+     */
+    public Mono<RestResponse<Secret>> recoverDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.recoverDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Backs up the specified secret.
+     * Requests that a backup of the specified secret be downloaded to the client. All versions of the secret will be downloaded.
+     * This operation requires the secrets/backup permission.
```

Comment:
Not a fan of the JavaDoc pattern you've got going on - the first sentence seems redundant and better explained by the second sentence?

Reply:
Updated.
Removed the first sentence, kept the second sentence.